events/source.go

```diff
@@ -94,6 +94,12 @@ func (es *Source) KeyChord(rn rune, code key.Codes, mods key.Modifiers) {
 // MouseButton creates and sends a mouse button event with given values
 func (es *Source) MouseButton(typ Types, but Buttons, where image.Point, mods key.Modifiers) {
 	ev := NewMouse(typ, but, where, mods)
+	if es.Last.MouseButtonType == MouseDown {
```

Comment:
shouldn't this be conditional on Up?  you definitely don't want to just have all down events be the same!?

Reply:
What do you mean? How does this make them all the same?

Reply:
You are copying from last always.

Reply:
I am just copying the start and prev info, which would otherwise be unset.

Reply:
Down should not have that info - only up.

Reply:
Done.